test/workload-deployment.sh

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env bash
+# Copyright 2024 The KubeStellar Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This is an end to end test of multi cluster deployement.  
+# For readable instructions, please visit https://github.com/kubestellar/kubestellar/tree/ks-0.20/docs/content/v0.20
+
+set -e # exit on error
+
+echo "Install KubeFlex"
+echo "-------------------------------------------------------------------------"
+OS_ARCH=linux_amd64
+LATEST_RELEASE_URL=$(curl -H "Accept: application/vnd.github.v3+json"   https://api.github.com/repos/kubestellar/kubeflex/releases/latest   | jq -r '.assets[] | select(.name | test("'${OS_ARCH}'")) | .browser_download_url')
+curl -LO $LATEST_RELEASE_URL
+mkdir -p kubeflex
+tar xzvf $(basename $LATEST_RELEASE_URL) -C kubeflex
+rm $(basename $LATEST_RELEASE_URL)
+export PATH=$PWD/kubeflex/bin:$PATH
+
+echo "Create a Kind hosting cluster with nginx ingress controller and KubeFlex operator"
+echo "-------------------------------------------------------------------------"
+kflex init --create-kind
+
+echo "Create an inventory & mailbox space of type vcluster running OCM (Open Cluster Management) directly in KubeFlex. Note that -p ocm runs a post-create hook on the vcluster control plane which installs OCM on it."
+echo "-------------------------------------------------------------------------"
+kflex create imbs1 --type vcluster -p ocm
+
+echo "Create a Workload Description Space wds1 directly in KubeFlex. Similarly to before, -p kslight runs a post-create hook on the k8s control plane that starts an instance of a KSLight controller manager which connects to the wds1 front-end and the imbs1 OCM control plane back-end."
+echo "-------------------------------------------------------------------------"
+kflex create wds1 -p kslight
+
+echo "Register clusters with OCM"
```

Comment:
This `echo` grossly understates what is being done here.

Reply:
updated.